ipfsutils/serviceMetadata.go

```diff
@@ -0,0 +1,98 @@
+package ipfsutils
+
+import (
+	"encoding/json"
+	"github.com/singnet/snet-daemon/config"
+	"math/big"
+	"strings"
+)
+
+type service_metadata struct {
+	Version                    int    `json:"version"`
+	DisplayName                string `json:"display_name"`
+	Encoding                   string `json:"encoding"`
+	ServiceType                string `json:"service_type"`
+	PaymentExpirationThreshold int    `json:"payment_expiration_threshold"`
+	ModelIpfsHash              string `json:"model_ipfs_hash"`
+	MpeAddress                 string `json:"mpe_address"`
+	Pricing                    struct {
+		PriceModel  string   `json:"price_model"`
+		PriceInCogs *big.Int `json:"price_in_cogs"`
+	} `json:"pricing"`
+	Groups []struct {
+		GroupName      string `json:"group_name"`
+		GroupID        string `json:"group_id"`
+		PaymentAddress string `json:"payment_address"`
+	} `json:"groups"`
+	Endpoints []struct {
+		GroupName string `json:"group_name"`
+		Endpoint  string `json:"endpoint"`
+	} `json:"endpoints"`
+	DeamonReplicaGroupID string
+	DeamonGroupName      string
+	DaemonEndPoint       string
+}
+
+var metaData *service_metadata
```

Comment:
Refactoring proposal (discussable and definitely can be postponed).

I would have ```service_metadata``` instance as a component instead of making global variable. It can be instantiated as a component in ```components.go``` and injected as dependency into all components which require it. It allows:
- having dependency graph in explicit form in ```components.go```; at the moment metadata is implicitly initialized in blockchain component and if there is a component which requires metadata but not requires blockchain it is not obvious that it should be initialized after blockchain;
- passing custom formed objects as parameters of the components in unit tests instead of setting global settings;